nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/service/client/TlsCertificateAuthorityClientCommandLine.java

```diff
@@ -0,0 +1,125 @@
+        addOptionNoArg("S", SAME_KEY_AND_KEY_STORE_PASSWORD_ARG, "When generating passwords, use the same one for KeyStore and Key");
+    }
+
+    public static void main(String[] args) throws Exception {
+        TlsHelper.addBouncyCastleProvider();
+        TlsCertificateAuthorityClientCommandLine tlsCertificateAuthorityClientCommandLine = new TlsCertificateAuthorityClientCommandLine();
+        try {
+            tlsCertificateAuthorityClientCommandLine.parse(args);
+        } catch (CommandLineParseException e) {
+            System.exit(e.getExitCode());
+        }
+        new TlsCertificateAuthorityClient().generateCertificateAndGetItSigned(tlsCertificateAuthorityClientCommandLine.createClientConfig(),
+                tlsCertificateAuthorityClientCommandLine.getCertificateDirectory(), tlsCertificateAuthorityClientCommandLine.getConfigJson(),
+                tlsCertificateAuthorityClientCommandLine.sameKeyAndKeyStorePassword());
+        System.exit(ExitCode.SUCCESS.ordinal());
+    }
+
+    @Override
+    protected boolean shouldAddDaysArg() {
+        return false;
+    }
+
+    @Override
+    protected boolean shouldAddSigningAlgorithmArg() {
+        return false;
+    }
+
+    @Override
+    protected String getKeyStoreTypeDefault() {
+        return PKCS_12;
+    }
+
+    @Override
+    protected CommandLine doParse(String[] args) throws CommandLineParseException {
+        CommandLine commandLine = super.doParse(args);
+        certificateDirectory = commandLine.getOptionValue(CERTIFICATE_DIRECTORY, DEFAULT_CERTIFICATE_DIRECTORY);
+        sameKeyAndKeyStorePassword = commandLine.hasOption(SAME_KEY_AND_KEY_STORE_PASSWORD_ARG);
+        return commandLine;
+    }
+
+    public boolean sameKeyAndKeyStorePassword() {
+        return sameKeyAndKeyStorePassword;
+    }
+
+    public String getCertificateDirectory() {
+        return certificateDirectory;
+    }
+
+    public TlsClientConfig createClientConfig() throws IOException {
+        if (onlyUseConfigJson()) {
+            try (InputStream inputStream = inputStreamFactory.create(new File(getConfigJson()))) {
+                TlsClientConfig tlsClientConfig = new ObjectMapper().readValue(inputStream, TlsClientConfig.class);
+                tlsClientConfig.initDefaults();
+                return tlsClientConfig;
+            }
+        } else {
+            TlsClientConfig tlsClientConfig = new TlsClientConfig();
+            tlsClientConfig.setCaHostname(getCertificateAuthorityHostname());
+            tlsClientConfig.setDn(getDn());
+            tlsClientConfig.setToken(getToken());
+            tlsClientConfig.setPort(getPort());
+            tlsClientConfig.setKeyStore(KEYSTORE + getKeyStoreType().toLowerCase());
+            tlsClientConfig.setKeyStoreType(getKeyStoreType());
+            tlsClientConfig.setTrustStore(TRUSTSTORE + getKeyStoreType().toLowerCase());
```

Comment:
This gets the default keystore type (`PKCS12`) to use for the truststore as well. The truststore rarely ever needs to be `PKCS12` as there is no private key information in it. I am not sure why the default (even for keystores) is not `JKS`?

Example:

```bash
hw12203:...assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT (pr695) alopresto
🔓 113s @ 16:01:50 $ ./bin/tls-toolkit.sh client -D CN=client.nifi.apache.org -t shorttoken
log4j:WARN No appenders could be found for logger (org.apache.http.client.protocol.RequestAddCookies).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
hw12203:...assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT (pr695) alopresto
🔓 43s @ 16:02:33 $ tl
.
...
├── [ 520]  config.json
├── [3.3K]  keystore.pkcs12
...
├── [2.1K]  nifi-ca-keystore.jks
├── [1.1K]  nifi-cert
├── [1.2K]  rootCert.crt
├── [1.6K]  rootCert.key
└── [1.1K]  truststore.pkcs12

3 directories, 36 files
```